app/concerns/long_runnable.rb

```diff
@@ -90,14 +92,21 @@ def stop!
       if respond_to?(:stop)
         stop
       else
-        thread.terminate
+        terminate_thread!
       end
     end
 
+    def terminate_thread!
+      thread.terminate
+      thread.wakeup if thread && thread.status == 'sleep'
```

Comment:
Thanks, I was mislead by the specs, checking the status is enough to ensure not to wakeup a killed thread.